OpenRA.Mods.Common/Traits/Infantry/TerrainModifiesDamage.cs

```diff
@@ -0,0 +1,61 @@
+﻿ #region Copyright & License Information
+ /*
+  * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+  * This file is part of OpenRA, which is free software. It is made
+  * available to you under the terms of the GNU General Public License
+  * as published by the Free Software Foundation. For more information,
+  * see COPYING.
+  */
+ #endregion
+
+using System;
+using System.Linq;
+using OpenRA.GameRules;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common
+{
+	public class TerrainModifiesDamageInfo : ITraitInfo
+	{
+		[Desc("An array of terrain types that modify this actor's incoming damage, if the actor is standing on these types.")]
+		public readonly string[] TerrainTypes = { };
+
+		[Desc("Total damage percent for each TerrainType. 120 = 120%, 80 = 80%, 50 = 50%, etc.")]
+		public readonly int[] DamagePerType = { };
```

Comment:
Not really a fan of these 2 flat arrays. I think a setup like MobileInfo's TerrainSpeeds would be nicer.